ui/src/views/WorkerManager/WMViewWorkerPools/index.jsx

```diff
@@ -35,6 +38,17 @@ export default class WorkerManagerWorkerPoolsView extends Component {
     this.props.history.push(`${this.props.match.path}/create`);
   };
 
+  deleteRequest = async ({ workerPoolId, payload }) => {
+    await this.props.client.mutate({
+      mutation: deleteWorkerPoolQuery,
+      variables: {
+        workerPoolId,
+        payload,
+      },
+    });
+    window.location.reload();
```

Comment:
I tried that - it doesn't work the way I want it to

Reply:
note to self: try this again to make sure

Reply:
ok, confirmed not working. Code for the function I tried:

```javascript
deleteRequest = async ({ workerPoolId, payload }) => {
    await this.props.client.mutate({
      mutation: updateWorkerPoolQuery,
      variables: {
        workerPoolId,
        payload: {
          ...payload,
          providerId: 'null-provider', // this is how we delete worker pools
        },
      },
      refetchQueries: ['workerPools'],
      awaitRefetchQueries: true,
    });
    // window.location.reload();
  };
```

The delete request itself is successful, but no refresh is happening

Reply:
The fetch happens as well, as I can tell from the Network tab. React tab confirms that the refetched array of worker pools is in props as it should be. No rerendering

Reply:
> React tab confirms that the refetched array of worker pools is in props as it should be. No rerendering

Ah. In that case, it's probably re-rendering but I suspect the serializer function (cache key) for `sortWorkerPools` doesn't change hence showing the same table. The serializer function should probably be amended to to return a new key when a mutation operation happens (e.g., delete). If changing the serializer function is not so straight forward for a mutation that is "scheduled", then we could probably remove the `memoize` feature.

Reply:
I'm not following. `sortWorkerPools` is called in `render` function which doesn't get called because there's no re-rendering for some reason (even despite the change in props).... what you're describing seems more like a symptom to me rather than the reason

Reply:
Interesting. Just so I understand, are you suggesting that the render function of `WMViewWorkerPools/index.jsx` runs again but the render function of the component children are not triggered?

Few things running in my head:
* The prop may be changing but the object reference may be the same which might cause children components not to re-render.
* We can try adding a `this.setState({ foo: 'bar' });` statement after `await this.props.client.mutate` just to see if this fixes anything. I'm not saying this is the solution but it is worth seeing if that causes a different behavior. 

Note: I don't have a way to test this since I can't login to the bstack deployment. I probably need access to login to the aws account and things like that. I can ask bstack to help with the configuration if the problem persists after trying to debug here.

Reply:
Ok, so I debugged this - the problem, indeed, was in the sorting function. The serializer wasn't working _at all_ because the sorting part was a bunch of `undefined`s - I decided not to bother to debug that and remove all the fancy `pipe`s etc. and just use plain good old javascript. I also added the provider ids to the string so that the memoize would re-cache everything if any provider id changes. It works for me now